ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/hateoas/HateoasMapperImpl.java

```diff
@@ -81,8 +65,28 @@ public void setOpenApiLinks(ServiceDeclarationNode serviceNode, OpenAPI openAPI)
             if (responses.isEmpty()) {
                 continue;
             }
-            setOpenApiLinksInApiResponse(serviceId, resource, responses.get());
+            setOpenApiLinksInApiResponse(hateoasService, resource, responses.get());
+        }
+    }
+
+    private Service extractHateoasMetaInfo(ServiceDeclarationNode serviceNode) {
+        Service service = new Service("ss", 0);
```

Comment:
Is there any reason for initializing the service object with a mock value?

Reply:
This has already been resolved by following commit,
- [4c8bdc3e56d26fd80f0fd8fce7c334516cfaa6c7](https://github.com/ballerina-platform/openapi-tools/pull/1589/commits/4c8bdc3e56d26fd80f0fd8fce7c334516cfaa6c7)